flaskr/blog.py

```diff
@@ -18,16 +19,24 @@
 def index():
     """Show all the posts, most recent first."""
     db = get_db()
+    page = request.args.get('page', 1, type=int)
+    per_page = 3
+    total_posts = db.execute("SELECT COUNT(*) FROM post").fetchone()[0]
+    total_pages = math.ceil(total_posts / per_page)
```

Comment:
**Issue**: Handle the case where `total_posts` is zero to avoid division by zero errors when calculating `total_pages`. Consider setting `total_pages` to 1 if `total_posts` is zero.

**Fix**: total_pages = math.ceil(total_posts / per_page) if total_posts > 0 else 1